pkg/arvo/app/dojo.hoon

```diff
@@ -1349,16 +1350,33 @@
   !>(state)
 ::
 ++  on-load
-  |=  =old-state=vase
-  =/  old-state  !<(house old-state-vase)
-  `..on-init(state old-state)
+  |=  old=vase
+  ?:  ?=(%6 +<.old)
+    `..on-init(state !<(house old))
+  =/  old-5  !<([%5 egg=@u hoc=(map id session)] old)
+  =/  =house  [%6 egg.old-5 hoc.old-5 *(set ship)]
+  `..on-init(state house)
```

Comment:
I hacked together this pretty sad looking state adapter -- feedback much appreciated

Reply:
This is basically correct, certainly fine for a single upgrade.  More complex ones are usually structured like this:

https://github.com/urbit/urbit/blob/69b6495d3d4c8cd3d81c5842a2cc90bf8059d63c/pkg/arvo/sys/vane/gall.hoon#L1665-L1683

Reply:
Curious: How is it that Dojo didn't already have a state adapter, given that its state was on version 5? Do "builtin" apps not need full 0->1, 1->2, 2->3... state adapters because they're updated OTA?

Reply:
It was upgraded in earlier continuity eras, this is the first state adapter since this era started in December.  I guess we deleted the adapters but didn't reset the version number to 0.